mtags-java/src/main/scala/scala/meta/internal/pc/SourceJavaFileObject.scala

```diff
@@ -0,0 +1,21 @@
+package scala.meta.internal.pc
+
+import java.io.File
+import java.net.URI
+import javax.tools.JavaFileObject.Kind
+import javax.tools.SimpleJavaFileObject
+
+class SourceJavaFileObject(src: String, uri: URI, kind: Kind)
+    extends SimpleJavaFileObject(uri, kind) {
+  override def getCharContent(ignoreEncodingErrors: Boolean): CharSequence = src
+}
+
+object SourceJavaFileObject {
+  def make(code: String): SourceJavaFileObject = {
+    val out = new StringBuilder()
+    out.append(code)
+
+    val tmpFile = File.createTempFile("code-", ".java")
+    new SourceJavaFileObject(code, URI.create(tmpFile.getPath), Kind.SOURCE)
+  }
```

Comment:
There is no need to create temp file.
Most PC methods receives `OffsetParams` which is already has `uri` value, so you need to pass the actual one. Otherwise, java compiler might produce a set of errors like: 
```
/tmp/code-14201405053267210121.java:35: error: class PresentationCompiler is public, should be declared in a file named PresentationCompiler.java
public abstract class PresentationCompiler {
```